docs/tutorial/types_structs_padding.md

```diff
@@ -0,0 +1,148 @@
+# Types, structs, and padding
+
+Reminders:
+- In N64 MIPS, 1 word is 4 bytes (yes, the N64 is meant to be 64-bit, but it mostly isn't used like it in OoT)
+- A byte is 8 bits, or 2 hex digits
+
+
+## Types
+
+The following are the common data types used everywhere:
+
+| Name  | Size            | Comment        |
+| ----  | -----           | --------       |
+| char  | 1 byte          | character      |
+| u8    | 1 byte          | unsigned byte  |
+| s8    | 1 byte          | signed byte    |
+| u16   | 2 bytes         | unsigned short | 
+| s16   | 2 bytes         | signed short   |
+| u32   | 4 bytes/1 word  | unsigned int   |
+| s32   | 4 bytes/1 word  | signed int     |
+| void* | 4 bytes/1 word  | pointer        |
+
+A pointer is sometimes mistaken for an s32. s32 is the default thing to use in the absence of any other information about the data.
+
+Useful data for guessing types:
+- u8 is about 7 times more common than s8 (so the tutorial is a bit misleading)
+- s16 is about 16 times more common than u16
+- s32 is about 8 times more common than u32
+
+Another useful thing to put here: the prototype for an action function is
+```C
+typedef void (*ActorNameActionFunc)(struct ActorName*, GlobalContext*);
+```
+where you replace `ActorName` by the actual actor name as used elsewhere in the actor, e.g. `EnJj`.
+
+
+## Some Common Structs
+
+
+
+Here are the usual names and the sizes of some of the most common structs used in actors and their structs:
+| Type                 | Usual name            | Size            |
+| -------------------- | --------------------- | --------------- |
+| `Actor`              | `actor`               | 0x14C           |
+| `DynaPolyActor`      | `dyna`                | 0x164           |
+| `Vec3f`              |                       | 0xC             | 
+| `Vec3s`              |                       | 0x6             | 
+| `SkelAnime`          | `skelAnime`           | 0x44            |
+| `Vec3s[limbCount]`   | `limbDrawTable`       | 0x6 * limbCount |
+| `Vec3s[limbCount]`   | `transitionDrawTable` | 0x6 * limbCount |
+| `ColliderCylinder`   | `collider`            | 0x4C            |
+| `ColliderQuad`       | `collider`            | 0x80            |
+| `ColliderJntSph`     | `collider`            | 0x20            |
+| `ColliderJntSphItem` | `colliderItems[n]`    | 0x40 * n        |
+| `ColliderTris`       | `collider`            | 0x20            |
+| `ColliderTrisItem`   | `colliderItems[n]`    | 0x5C * n        |
```

Comment:
same comment about elements from earlier

Reply:
Shall I change the default name too? I see it hasn't been batch-renamed in the repo yet.